### PR TITLE
Support for image Views.

### PR DESCRIPTION
**Pull request by qtzar**
This PR introduces support for Image Views.

_An [image view](https://structurizr.com/help/image-views) represents an image that has been rendered outside of Structurizr, for inclusion in a Structurizr workspace. Image views can be created from an arbitrary PNG/SVG file, or a PlantUML/Mermaid/Kroki diagram. An example use case for an image view is to provide a “level 4” zoom-in for a component on a component diagram._

From experimentation you can associate an image view with a SoftwareSystem, Container or Component and you can have multiple images views associated against a single element.

- [x] Show Image Views for SoftwareSystems on the Containers View page.
- [x] Show Image Views for Containers on the Components View page.
- [x] Show Image Views for Components on a new 'Code Views' page.
- [x] Make sure SVG links are setup to go to those pages even if only an image view exists on the page.
- [x] Write appropriate tests

### CODE REVIEW

**Comment by qtzar** [comment]
@dirkgroot @jp7677 This PR might take me a while to complete. Please don't delay a new release waiting for it to be completed.

File: src/main/kotlin/nl/avisi/structurizr/site/generatr/StructurizrUtilities.kt [hunk 1]
```diff
@@ -36,14 +38,43 @@ fun Container.hasSections() = documentation.sections.isNotEmpty()
...
 fun ViewSet.hasDeploymentViews(softwareSystem: SoftwareSystem) =
     deploymentViews.any { it.softwareSystem == softwareSystem }
+
+fun getImagesForSystem(generatorContext: GeneratorContext, softwareSystem: SoftwareSystem): List<ImageView> {
+    return generatorContext.workspace.views.imageViews
+        .filter { it.elementId == softwareSystem.id }
+        .sortedBy { it.key }
+}
+
+fun getImagesForContainer(generatorContext: GeneratorContext, softwareSystem: SoftwareSystem): List<ImageView> {
```
Comment (qtzar) [inline src/main/kotlin/nl/avisi/structurizr/site/generatr/StructurizrUtilities.kt:62]: @jp7677 Is there a better way to write this in Kotlin?

Comment (jp7677) [reply src/main/kotlin/nl/avisi/structurizr/site/generatr/StructurizrUtilities.kt]: From a short glimpse, Kotlin's `flatMap` should be your friend here https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/flat-map.html

**Comment by jp7677** [comment]
Thanks a lot for the PR, I haven't yet looked at the code since I want to complete https://github.com/avisi-cloud/structurizr-site-generatr/pull/356 first. That said I have checked out this branch and looked at the result. Looks good and fits nicely. A few remarks:

- I would prefer if we show images the same way we do with diagrams, thus
  -  the same figure construction
  - the same place for name and description. We don't show the description yet for diagrams. So for now no description yet for images,  or also adding the description for diagrams
  - We should also add the modal dialog for "opening" image, albeit no pan and zoom obviously. 
- Please don't modify the example that heavily, at least not as part of this PR, it's fine of course to add the images
- One of the sample images has a thick boarder, could you find an alternative that fits nicer in the resulting example site?

**Comment by jp7677** [comment]
Hi @qtzar do you have plans to continue this PR?

**Comment by qtzar** [comment]
@jp7677 I do. I've just been up to my eyes with work but I should get it finished off.

**Comment by jp7677** [comment]
> @jp7677 I do. I've just been up to my eyes with work but I should get it finished off.

Cool and no worries about timelines, I was just curious if you are still interested to continue.

**Comment by qtzar** [comment]
I've updated this branch to work with the latest release but I still need to address some of the comments from @jp7677

**Comment by jp7677** [comment]
I've opened https://github.com/avisi-cloud/structurizr-site-generatr/pull/413 which takes most from this PR, but with slightly different styling and more tests. Thanks a lot for the base work here!

**Pull request closed without merging**